### PR TITLE
Admin panel RBAC. New admin roles support

### PR DESCRIPTION
**Pull request by mnaichuk**

### CODE REVIEW

File: app/models/abilities/accountant_ability.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# encoding: UTF-8
+# frozen_string_literal: true
+
+module Abilities
+  class AccountantAbility
```
Comment (mnaichuk) [inline app/models/abilities/accountant_ability.rb]: lets rename this classes to simple Admin, Accountant and etc..

File: app/models/abilities/superadmin_ability.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      can :manage, Operation
+      can :manage, Market
+      can :manage, Currency
+      can :manage, Blockchain
+      can :manage, Wallet
+      can :manage, Account
+      can :manage, PaymentAddress
+    end
+  end
+end
```
Comment (mnaichuk) [inline app/models/abilities/superadmin_ability.rb]: add empty line

**Comment by mnaichuk** [comment]
Add specs for each role please

**Comment by ysv** [comment]
We need to commit table with user roles and permissions either in md or directly inside `module Abilities`

File: app/views/admin/deposits/fiats/index.html.erb [hunk 3]
```diff
@@ -1,27 +1,31 @@
...
-  <div class="float-right">
-    <%= paginate @all_deposits, params: { route: :admin_deposit_index } %>
+<% if can? :read, Deposit %>
+  <div class="row">
+    <div class="col-6">
+      <%= link_to raw("All deposits"), admin_deposit_index_path(state: 'all'), class: 'btn btn-primary  mb-3' %>
+      <%= link_to raw("Latest deposits"), admin_deposit_index_path(state: 'latest'), class: 'btn btn-primary  mb-3' %>
+    </div>
+    <div class="col-6">
+      <% if can? :write, Deposit %>
```
Comment (mnaichuk) [inline app/views/admin/deposits/fiats/index.html.erb]: can? :manage

**Review by mnaichuk — COMMENTED** [review]
Add can? for members view

File: app/views/admin/deposits/fiats/new.html.erb [hunk 4]
```diff
@@ -22,6 +22,7 @@
     </div>
   </div>
 
-  <%= f.submit 'Accept', class: 'btn btn-primary btn-lg mt-3' %>
-
+  <% if can? :write, Deposit %>
```
Comment (mnaichuk) [inline app/views/admin/deposits/fiats/new.html.erb]: :manage

**Review by mnaichuk — COMMENTED** [review]
also check ./app/views/shared/admin

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -17,6 +17,7 @@
 require 'rspec/rails'
 require 'rspec/retry'
 require 'webmock/rspec'
+require "cancan/matchers"
```
Comment (ysv) [inline spec/spec_helper.rb]: use single quotes here

File: app/models/abilities.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# encoding: UTF-8
+# frozen_string_literal: true
+
+module Abilities
```
Comment (ysv) [inline app/models/abilities.rb:5]: Add comment that full list of roles abilities could be found on docs/roles.md

**Review by ysv — CHANGES_REQUESTED** [review]

**Review by ysv — APPROVED** [review]

**Comment by pkucherenk0** [comment]
As far as i know we already have member with role broker and trades on barong, same permissions as a member. This members just a preparation for upcoming possibility's, but we should add them anyway.

**Review by ysv — APPROVED** [review]

File: app/models/abilities/support.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
+# encoding: UTF-8
+# frozen_string_literal: true
+
+module Abilities
+  class Support
+    include CanCan::Ability
+
+    def initialize(user)
+      can :read, Operations::Account
```
Comment (mod) [inline app/models/abilities/support.rb]: Support cannot see low level accounting

File: app/models/abilities/support.rb [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+      can :read, Operations::Asset
+      can :read, Operations::Expense
+      can :read, Operations::Liability
+      can :read, Operations::Revenue
+
+      can :read, Member
+      can :read, Deposit
+      can :read, Withdraw
+      can :read, Account
+      can :read, PaymentAddress
```
Comment (mod) [inline app/models/abilities/support.rb]: Dont show this one

File: app/models/abilities/technical.rb [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+module Abilities
+  class Technical
+    include CanCan::Ability
+
+    def initialize(user)
+      can :manage, Market
+      can :manage, Currency
+      can :manage, Blockchain
+      can :manage, Wallet
```
Comment (mod) [inline app/models/abilities/technical.rb:18]: can :read, Member
      can :read, Deposit
      can :read, Withdraw
      can :read, Account
      can :read, PaymentAddress

File: app/models/abilities/accountant.rb [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+module Abilities
+  class Accountant
+    include CanCan::Ability
+
+    def initialize(user)
+      can :read, Operations::Account
+      can :read, Operations::Asset
+      can :read, Operations::Expense
+      can :read, Operations::Liability
+      can :read, Operations::Revenue
```
Comment (mod) [inline app/models/abilities/accountant.rb:13]: create deposit but not accept

**Review by mod — CHANGES_REQUESTED** [review]

**Pull request merged**